tools/build_api.py

```diff
@@ -230,11 +231,15 @@ def find_valid_toolchain(target, toolchain):
     """
     end_warnings = []
     toolchain_names = get_valid_toolchain_names(target, toolchain)
+    if target.default_toolchain == "uARM":
+        end_warnings.append(UARM_TOOLCHAIN_WARNING)
     last_error = None
     for index, toolchain_name in enumerate(toolchain_names):
         internal_tc_name = get_toolchain_name(target, toolchain_name)
         if toolchain == "ARM" and toolchain_name == "ARMC5" and index != 0:
             end_warnings.append(ARMC5_MIGRATION_WARNING)
+        elif toolchain_name == "uARM":
```

Comment:
What is the value of `toolchain` at this point? Why is it not necessary to also check `toolchain` as done in line `239`?